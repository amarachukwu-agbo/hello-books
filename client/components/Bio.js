import React from 'react';

const Bio = props => (
  <div>
    <div className="row">
      <div className="col s12 m6 l6">
        <p className="bold"><i className="material-icons prefix teal-text">account_circle</i><span className="bold blue-text">First Name:</span>
          <span> {props.bio.firstName}</span></p>
      </div>
      <div className="col s12 m6 l6">
        <p><span className="bold blue-text">LastName: </span><span> {props.bio.lastName}</span></p>
      </div>
    </div>
    <div className="row">
      <div className="col s12 m6 l6">
        <p className="bold"><i className="material-icons prefix teal-text">email</i><span className="bold blue-text">Email: </span>
          <span> {props.bio.email}</span></p>
      </div>
      <div className="col s12 m6 l6">
        <p><span className="bold blue-text">Member Since: </span><span> {props.bio.createdAt.split('T')[0]}</span></p>
      </div>
    </div>
  </div>
);

export default Bio;
