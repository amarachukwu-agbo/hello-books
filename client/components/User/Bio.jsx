import React from 'react';
import propTypes from 'prop-types';
import imagePlaceholder from '../../public/images/profile-placeholder.png';

/**
 * @description stateless component for rendering user's bio
 *
 * @param {object} props - properties of the component
 *
 * @returns {Node} - react node containing the Bio component
 */
const Bio = ({ bio }) => (
  <div>
    <div className="row">
      <div className="profile-img-div left">
        <img src={imagePlaceholder} className="profile-img responsive-img" />
      </div>
      <div className="bio-details left collection">
      <p className="collection-item">
        <strong className="bold primary-text">
          First Name: </strong>
        <span> {bio.firstName}</span>
      </p>
        <p className="collection-item">
          <strong className="bold primary-text">
            Last Name:
          </strong><span> {bio.lastName}</span>
        </p>
        <p className="collection-item">
          <strong className="bold primary-text">
            Email:
          </strong>
          <span> {bio.email}</span>
        </p>
        <p className="collection-item">
          <strong className="bold primary-text">
            Member Since:
          </strong>
          <span> {bio.createdAt.split('T')[0]}</span>
        </p>
      </div>
    </div>
  </div>
);

// Prop types validation
Bio.propTypes = {
  bio: propTypes.shape({
    firstName: propTypes.string.isRequired,
    lastName: propTypes.string.isRequired,
    createdAt: propTypes.string.isRequired,
    email: propTypes.string.isRequired,
  }),
};

export default Bio;
