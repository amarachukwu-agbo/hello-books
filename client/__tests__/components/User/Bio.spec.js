import React from 'react';
import { shallow } from 'enzyme';
import Bio from '../../../components/User/Bio';

const props = {
  bio: {
    firstName: 'Tester',
    lastName: 'User',
    email: 'tester@gmail.com',
    createdAt: '2018-05-10T03:11:52.181Z',
  },
};

describe('<Bio>', () => {
  it('should render user\'s bio correctly', () => {
    const tree = shallow(<Bio {...props}/>);
    expect(tree).toMatchSnapshot();
  });
});
