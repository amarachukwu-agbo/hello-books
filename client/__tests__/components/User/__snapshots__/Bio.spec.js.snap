// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bio> should render user's bio correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bio
    bio={
      Object {
        "createdAt": "2018-05-10T03:11:52.181Z",
        "email": "tester@gmail.com",
        "firstName": "Tester",
        "lastName": "User",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="profile-img-div left"
        >
          <img
            className="profile-img responsive-img"
            src="test-file-stub"
          />
        </div>
        <div
          className="bio-details left collection"
        >
          <p
            className="collection-item"
          >
            <strong
              className="bold primary-text"
            >
              First Name: 
            </strong>
            <span>
               
              Tester
            </span>
          </p>
          <p
            className="collection-item"
          >
            <strong
              className="bold primary-text"
            >
              Last Name:
            </strong>
            <span>
               
              User
            </span>
          </p>
          <p
            className="collection-item"
          >
            <strong
              className="bold primary-text"
            >
              Email:
            </strong>
            <span>
               
              tester@gmail.com
            </span>
          </p>
          <p
            className="collection-item"
          >
            <strong
              className="bold primary-text"
            >
              Member Since:
            </strong>
            <span>
               
              2018-05-10
            </span>
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="profile-img-div left"
          >
            <img
              className="profile-img responsive-img"
              src="test-file-stub"
            />
          </div>,
          <div
            className="bio-details left collection"
          >
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                First Name: 
              </strong>
              <span>
                 
                Tester
              </span>
            </p>
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                Last Name:
              </strong>
              <span>
                 
                User
              </span>
            </p>
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                Email:
              </strong>
              <span>
                 
                tester@gmail.com
              </span>
            </p>
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                Member Since:
              </strong>
              <span>
                 
                2018-05-10
              </span>
            </p>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="profile-img responsive-img"
              src="test-file-stub"
            />,
            "className": "profile-img-div left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "profile-img responsive-img",
              "src": "test-file-stub",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  First Name: 
                </strong>
                <span>
                   
                  Tester
                </span>
              </p>,
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  Last Name:
                </strong>
                <span>
                   
                  User
                </span>
              </p>,
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  Email:
                </strong>
                <span>
                   
                  tester@gmail.com
                </span>
              </p>,
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  Member Since:
                </strong>
                <span>
                   
                  2018-05-10
                </span>
              </p>,
            ],
            "className": "bio-details left collection",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong
                    className="bold primary-text"
                  >
                    First Name: 
                  </strong>,
                  <span>
                     
                    Tester
                  </span>,
                ],
                "className": "collection-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "First Name: ",
                    "className": "bold primary-text",
                  },
                  "ref": null,
                  "rendered": "First Name: ",
                  "type": "strong",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Tester",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Tester",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong
                    className="bold primary-text"
                  >
                    Last Name:
                  </strong>,
                  <span>
                     
                    User
                  </span>,
                ],
                "className": "collection-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Last Name:",
                    "className": "bold primary-text",
                  },
                  "ref": null,
                  "rendered": "Last Name:",
                  "type": "strong",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "User",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "User",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong
                    className="bold primary-text"
                  >
                    Email:
                  </strong>,
                  <span>
                     
                    tester@gmail.com
                  </span>,
                ],
                "className": "collection-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email:",
                    "className": "bold primary-text",
                  },
                  "ref": null,
                  "rendered": "Email:",
                  "type": "strong",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "tester@gmail.com",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "tester@gmail.com",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong
                    className="bold primary-text"
                  >
                    Member Since:
                  </strong>,
                  <span>
                     
                    2018-05-10
                  </span>,
                ],
                "className": "collection-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Member Since:",
                    "className": "bold primary-text",
                  },
                  "ref": null,
                  "rendered": "Member Since:",
                  "type": "strong",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "2018-05-10",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "2018-05-10",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="profile-img-div left"
          >
            <img
              className="profile-img responsive-img"
              src="test-file-stub"
            />
          </div>
          <div
            className="bio-details left collection"
          >
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                First Name: 
              </strong>
              <span>
                 
                Tester
              </span>
            </p>
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                Last Name:
              </strong>
              <span>
                 
                User
              </span>
            </p>
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                Email:
              </strong>
              <span>
                 
                tester@gmail.com
              </span>
            </p>
            <p
              className="collection-item"
            >
              <strong
                className="bold primary-text"
              >
                Member Since:
              </strong>
              <span>
                 
                2018-05-10
              </span>
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="profile-img-div left"
            >
              <img
                className="profile-img responsive-img"
                src="test-file-stub"
              />
            </div>,
            <div
              className="bio-details left collection"
            >
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  First Name: 
                </strong>
                <span>
                   
                  Tester
                </span>
              </p>
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  Last Name:
                </strong>
                <span>
                   
                  User
                </span>
              </p>
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  Email:
                </strong>
                <span>
                   
                  tester@gmail.com
                </span>
              </p>
              <p
                className="collection-item"
              >
                <strong
                  className="bold primary-text"
                >
                  Member Since:
                </strong>
                <span>
                   
                  2018-05-10
                </span>
              </p>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="profile-img responsive-img"
                src="test-file-stub"
              />,
              "className": "profile-img-div left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "profile-img responsive-img",
                "src": "test-file-stub",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="collection-item"
                >
                  <strong
                    className="bold primary-text"
                  >
                    First Name: 
                  </strong>
                  <span>
                     
                    Tester
                  </span>
                </p>,
                <p
                  className="collection-item"
                >
                  <strong
                    className="bold primary-text"
                  >
                    Last Name:
                  </strong>
                  <span>
                     
                    User
                  </span>
                </p>,
                <p
                  className="collection-item"
                >
                  <strong
                    className="bold primary-text"
                  >
                    Email:
                  </strong>
                  <span>
                     
                    tester@gmail.com
                  </span>
                </p>,
                <p
                  className="collection-item"
                >
                  <strong
                    className="bold primary-text"
                  >
                    Member Since:
                  </strong>
                  <span>
                     
                    2018-05-10
                  </span>
                </p>,
              ],
              "className": "bio-details left collection",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <strong
                      className="bold primary-text"
                    >
                      First Name: 
                    </strong>,
                    <span>
                       
                      Tester
                    </span>,
                  ],
                  "className": "collection-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "First Name: ",
                      "className": "bold primary-text",
                    },
                    "ref": null,
                    "rendered": "First Name: ",
                    "type": "strong",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Tester",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Tester",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <strong
                      className="bold primary-text"
                    >
                      Last Name:
                    </strong>,
                    <span>
                       
                      User
                    </span>,
                  ],
                  "className": "collection-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Last Name:",
                      "className": "bold primary-text",
                    },
                    "ref": null,
                    "rendered": "Last Name:",
                    "type": "strong",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "User",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "User",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <strong
                      className="bold primary-text"
                    >
                      Email:
                    </strong>,
                    <span>
                       
                      tester@gmail.com
                    </span>,
                  ],
                  "className": "collection-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email:",
                      "className": "bold primary-text",
                    },
                    "ref": null,
                    "rendered": "Email:",
                    "type": "strong",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "tester@gmail.com",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "tester@gmail.com",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <strong
                      className="bold primary-text"
                    >
                      Member Since:
                    </strong>,
                    <span>
                       
                      2018-05-10
                    </span>,
                  ],
                  "className": "collection-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Member Since:",
                      "className": "bold primary-text",
                    },
                    "ref": null,
                    "rendered": "Member Since:",
                    "type": "strong",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "2018-05-10",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "2018-05-10",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
